liquid-compiler/src/registry.rs

```diff
@@ -21,8 +21,8 @@ impl<P> PluginRegistry<P> {
     /// Generally this is used when setting up the program.
     ///
     /// Returns whether this overrode an existing plugin.
-    pub fn register(&mut self, name: &str, plugin: P) -> bool {
-        let old = self.plugins.insert(name.to_string(), plugin);
+    pub fn register(&mut self, name: String, plugin: P) -> bool {
+        let old = self.plugins.insert(name, plugin);
```

Comment:
I wish Rust had a Map where it would extract the key from the value.  With that, we wouldn't need to do any of this.